src/server/engine_shard.cc

```diff
@@ -704,6 +709,7 @@ void EngineShard::RetireExpiredAndEvict() {
   { std::unique_lock lk(db_slice.GetSerializationMutex()); }
   constexpr double kTtlDeleteLimit = 200;
   constexpr double kRedLimitFactor = 0.1;
```

Comment:
rename to kEvictionFreeMemoryThreshold
add a comment when shard free memory goes below max_memory*threshold we start evicting